demo-app/src/lib/synchronous-graphql-query.ts

```diff
@@ -0,0 +1,87 @@
+import sources from '@/lib/graphql-sources'
+import { GraphQLClient } from 'graphql-request'
+
+interface SynchronousGraphqlResult {
+  status: number
+  headers: Record<string, string[]>
+  body: any
+  errors: any
+}
+
+interface Props {
+  source: keyof typeof sources
+  headers: Record<string, string>
+  body: string
+  variables: Record<string, any>
+}
+
+async function synchronousGraphqlQuery({
+  source,
+  headers,
+  body,
+  variables,
+}: Props): Promise<SynchronousGraphqlResult> {
+  const config = sources[source]?.config
+  if (!config) throw new Error(`No such source: ${source}`)
+
+  const { uri, headers: defaultHeaders } = config
+
+  // Use 'all' to get both data and errors
+  const graphClient = new GraphQLClient(uri, {
+    errorPolicy: 'all',
+    headers: {
+      ...defaultHeaders,
+      ...headers,
+    },
+  })
+
+  console.log({
+    query: body,
+    variables,
+  })
```

Comment:
Should this be wrapped in `_DEBUG_`? Akin to:
https://github.com/open-truss/open-truss/blob/00d1ed8275de4bed876ee5038615bf362c12ea0f/packages/open-truss/src/components/OTRestDataProvider.tsx#L38-L45